### ISSUE TITLE
Use of `memset` does not guarantee memory is zeroed prior to `free`.

### ISSUE THREAD

**Issue by TomMD** [linux-pam/linux-pam#305]
Motivated by mention at #303 (and follow on discussion in #304) I wrote[ a Cobra pattern](https://github.com/nimble-code/Cobra) to find all instances of memset/free (and suggest we update to memset_s/free).  With respect to #304, I will get this in Muse as time allows but didn't want to hold you up if it was on your radar.

The command is:

```
find . -name '*.c' | xargs \
    cobra --cpp -pat '{ .* memset ( .*x:@ident , .* , .* ) .* free ( .*:x ) .* } ' | \
    grep -Ev '^ ' | grep ':'
```

Yielding the start of the statement block for each detected memset/free invocation:

```
pam_handlers.c:1030:
hmacsha1.c:60:
pam_userdb.c:184:
pam_selinux.c:388:
mkhomedir_helper.c:36:
pam_xauth.c:354:
passverify.c:72:
bigcrypt.c:54:
pam_secret.c:100:
pamc_load.c:12:
pamc_client.c:109:
```

**Comment by TomMD** [comment]
The above pattern isn't quite right since `.*` can allow use of the variable `:x` before `free()`.  The correct pattern is:

```
{ .* memset ( ^)* x:@ident , ^)* , ^)* ) ^:x* free ( ^):x ) .* }
```

Which still yields interesting places in linux_pam that should be fixed.

**Comment by ldv-alt** [comment]
See #531.

**Comment by cgzones** [comment]
Should be addressed by #528.

**Comment by ldv-alt** [comment]
Addressed by #528.
